src/Checkout/AddressForm.js

```diff
@@ -0,0 +1,272 @@
+  }
+
+  const initialFormState = {
+    name: editAddress.name,
+    phoneNumber: editAddress.phoneNumber,
+    zipCode: editAddress.zipCode,
+    city: editAddress.city,
+    address: editAddress.address,
+    state: editAddress.state,
+    country: editAddress.country,
+    addressType: editAddress.addressType,
+  };
+
+  const [formState, formDispatch] = useReducer(formReducer, initialFormState);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (/\+?\d[\d -]{8,12}\d/.test(formState.phoneNumber)) {
+      if (/^[1-9][0-9]{5}$/.test(formState.zipCode)) {
+        try {
+          userDispatch({ type: "STATUS", payload: "Adding Address...." });
+          const response = await axios.post(
+            `https://api-prestore.prerananawar1.repl.co/user-details/address`,
+            {
+              newAddress: { ...formState },
+            }
+          );
+          if (response.status === 201) {
+            userDispatch({
+              type: "ADD_ADDRESS",
+              payload: response.data.address,
+            });
+          }
+          setMsg("");
+          setAddNewAddress(false);
+        } catch (error) {
+          userDispatch({
+            type: "STATUS",
+            payload: "Couldn't add address..",
+          });
+        } finally {
+          userDispatch({ type: "STATUS", payload: "" });
+        }
+      } else {
+        setMsg("Please enter a valid Indian Zip-Code");
+      }
+    } else {
+      setMsg("Please enter a valid 10 digit phone number");
+    }
+  };
+
+  const handleEditAddress = async (e) => {
+    e.preventDefault();
+    if (/\+?\d[\d -]{8,12}\d/.test(formState.phoneNumber)) {
+      if (/^[1-9][0-9]{5}$/.test(formState.zipCode)) {
+        try {
+          userDispatch({
+            type: "STATUS",
+            payload: "Adding Updated Address....",
+          });
+          const response = await axios.post(
+            `https://api-prestore.prerananawar1.repl.co/user-details/address/${editAddID}`,
+            {
+              updateAddress: { ...formState },
+            }
+          );
+          if (response.status === 200) {
+            userDispatch({
+              type: "EDIT_ADDRESS",
+              payload: { id: editAddID, address: response.data.address },
+            });
+            setMsg("");
+            setEditAdd(false);
+            setAddNewAddress(false);
+            formDispatch({
+              type: "RESET_FORM",
+              payload: initialFormState,
+            });
+          }
+        } catch (error) {
+          userDispatch({
+            type: "STATUS",
+            payload: "Couldn't update address..",
+          });
+        } finally {
+          userDispatch({ type: "STATUS", payload: "" });
+        }
+      } else {
+        setMsg("Please enter a valid Indian Zip-Code");
+      }
+    } else {
+      setMsg("Please enter a valid 10 digit phone number");
+    }
+  };
```

Comment:
you can extract this function in a separate file, it will make this file look cleaner